browser/ui/views/commands/accelerator_service.cc

```diff
@@ -0,0 +1,188 @@
+    command->accelerators.push_back(std::move(a));
+  }
+  return command;
+}
+
+base::flat_map<int, mojom::CommandPtr> ToMojoCommands(
+    base::flat_map<int, std::vector<ui::Accelerator>> commands) {
+  base::flat_map<int, mojom::CommandPtr> result;
+  for (const auto& [command_id, accelerators] : commands) {
+    result[command_id] = ToMojoCommand(command_id, accelerators);
+  }
+  return result;
+}
+}  // namespace
+
+AcceleratorService::AcceleratorService(PrefService* pref_service,
+                                       const Accelerators& default_accelerators)
+    : pref_manager_(pref_service), default_accelerators_(default_accelerators) {
+  Initialize();
+}
+
+AcceleratorService::~AcceleratorService() = default;
+
+void AcceleratorService::Initialize() {
+  auto accelerators = pref_manager_.GetAccelerators();
+  if (accelerators.size() == 0) {
+    for (const auto& [command_id, default_accelerators] :
+         default_accelerators_) {
+      for (const auto& default_accelerator : default_accelerators) {
+        pref_manager_.AddAccelerator(command_id, default_accelerator);
+      }
+    }
+    accelerators = pref_manager_.GetAccelerators();
+  }
+
+  auto commands = GetCommands();
+  for (const auto& command : commands) {
+    accelerators_.insert({command, {}});
+
+    auto it = accelerators.find(command);
+    if (it == accelerators.end()) {
+      continue;
+    }
+
+    // Initialize the default table.
+    for (const auto& accelerator : it->second) {
+      accelerators_[command].push_back(accelerator);
+    }
+  }
+}
+
+void AcceleratorService::BindInterface(
+    mojo::PendingReceiver<CommandsService> pending_receiver) {
+  receivers_.Add(this, std::move(pending_receiver));
+}
+
+void AcceleratorService::AssignAcceleratorToCommand(
+    int command_id,
+    const std::string& accelerator) {
+  auto a = FromCodesString(accelerator);
+
+  std::vector<int> modified_commands{command_id};
+
+  // Find any other commands with this accelerator and remove it from them.
+  for (auto& [other_command_id, accelerators] : accelerators_) {
+    if (base::Contains(accelerators, a)) {
+      base::Erase(accelerators, a);
+      pref_manager_.RemoveAccelerator(other_command_id, a);
+      modified_commands.push_back(other_command_id);
+    }
+  }
+
+  accelerators_[command_id].push_back(a);
+  pref_manager_.AddAccelerator(command_id, a);
+  NotifyCommandsChanged(modified_commands);
+}
+
+void AcceleratorService::UnassignAcceleratorFromCommand(
+    int command_id,
+    const std::string& accelerator) {
+  auto a = FromCodesString(accelerator);
+  base::Erase(accelerators_[command_id], a);
+  pref_manager_.RemoveAccelerator(command_id, a);
+  NotifyCommandsChanged({command_id});
+}
+
+void AcceleratorService::ResetAcceleratorsForCommand(int command_id) {
+  std::vector<int> modified_commands{command_id};
+
+  // First, clear our shortcuts list.
+  auto& current_shortcuts = accelerators_[command_id];
+  current_shortcuts.resize(0);
```

Comment:
optional: `current_shortcuts.clear()` would be more intuitive.

Reply:
:facepalm: I was racking my brain trying to remember what you told me it was last time.